packages/mobx/src/api/tojs.ts

```diff
@@ -49,10 +50,11 @@ function toJSHelper(source, __alreadySeen: Map<any, any>) {
         return res
     } else {
         // must be observable object
-        keys(source) // make sure keys are observed
         const res = cache(__alreadySeen, source, {})
-        getPlainObjectKeys(source).forEach((key: any) => {
-            res[key] = toJSHelper(source[key], __alreadySeen)
+        ;((source as any) as IIsObservableObject)[$mobx].ownKeys_().forEach((key: any) => {
```

Comment:
I think the fix here is incorrect; we should use `keys(source)` as basis for the loop, and _not_ use `getPlainObjectKeys`, to make sure that `toJS` will track key additions, even when it is used in ES5 mode. There are probably subtle differences between `keys` and `getPlainObjectKeys` that might need to be addressed, but we should use `keys` as basis, or keep it as implemented here, but make sure a read to the keys atom is registered.

Reply:
`adm.ownKeys_` handles the subscription to keys atom the same way as `keys(object)`. We simply don't have `ownKeys_` exposed via `object-api` so I call it directly on administration.

Reply:
Just to clarify further, there is no longer any interesting logic (like keyAtom subscriptions) in object-api or in proxy handler. They simply delegates to `adm` and everything is done there.

Reply:
ok great, that should work!

Reply:
Leaving the merging up to you, in case you want to merge the open PR's in a specific order :)